File: mellsell-frontend/src/services/reviewService.js
import api from './api'

const listByProduct = (productId, page = 0, size = 20) => api.get(`/api/reviews/product/${productId}`, { params: { page, size } }).then(r => r.data)
const addReview = (payload) => api.post('/api/reviews', payload).then(r => r.data)

// admin endpoints
const listPending = (page = 0, size = 20) => api.get(`/api/admin/reviews/pending`, { params: { page, size } }).then(r => r.data)
const approveReview = (id) => api.post(`/api/admin/reviews/${id}/approve`).then(r => r.data)
const rejectReview = (id) => api.post(`/api/admin/reviews/${id}/reject`).then(r => r.data)

export default { listByProduct, addReview, listPending, approveReview, rejectReview }

File: mellsell-backend/src/main/java/com/mellsell/review/service/ReviewService.java
package com.mellsell.review.service;

import com.mellsell.auth.entity.User;
import com.mellsell.review.dto.CreateReviewRequest;
import com.mellsell.review.dto.ReviewResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReviewService {
    ReviewResponseDTO addReview(User user, CreateReviewRequest req);
    Page<ReviewResponseDTO> listByProduct(Long productId, Pageable pageable);

    // Admin actions
    Page<ReviewResponseDTO> listPending(Pageable pageable);
    ReviewResponseDTO approveReview(Long reviewId);
    ReviewResponseDTO rejectReview(Long reviewId);
}
